pkg/models/iam/im.go

```diff
@@ -883,6 +884,28 @@ func UpdateUser(user *models.User) (*models.User, error) {
 	dn := fmt.Sprintf("uid=%s,%s", user.Username, ldapclient.UserSearchBase)
 	userModifyRequest := ldap.NewModifyRequest(dn, nil)
 	if user.Email != "" {
+		userSearchRequest := ldap.NewSearchRequest(
+			ldapclient.UserSearchBase,
+			ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+			fmt.Sprintf("(&(objectClass=inetOrgPerson)(mail=%s))", user.Email),
+			[]string{"uid", "mail"},
+			nil,
+		)
+		result, err := conn.Search(userSearchRequest)
+		if err != nil {
+			glog.Error(err)
+			return nil, err
+		}
+		if len(result.Entries) > 1 {
+			err = ldap.NewError(ldap.ErrorDebugging, fmt.Errorf("email is duplicate"))
```

Comment:
I think we'd better have email in error message, something like
`fmt.Errorf(duplicated email: %s, email)`